client/config/config.go

```diff
@@ -57,6 +57,11 @@ var (
 		"/run/resolvconf": "/run/resolvconf",
 		"/sbin":           "/sbin",
 		"/usr":            "/usr",
+
+		// embed systemd-resolved paths for systemd-resolved paths:
+		// /etc/resolv.conf is a symlink to /run/systemd/resolve/stub-resolv.conf in such systems.
+		// In non-systemd systems, this mount is a no-op and the path is ignored if not present.
+		"/run/systemd/resolve": "/run/systemd/resolve",
```

Comment:
Note that in https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html#/etc/resolv.conf , it notes the possibility of symlinking to /usr/lib/systemd/resolv.conf but I haven't run into this, and docker doesn't seem to attempt to handle that case.